src/journal/src/Journal/Internal/ByteBuffer.hs

```diff
@@ -243,41 +272,120 @@ getStorable bb = do
 
 putStorableAt :: Storable a => ByteBuffer -> Int -> a -> IO ()
 putStorableAt bb ix x = do
-  boundCheck bb ix
+  boundCheck bb "putStorableAt" ix
   pokeByteOff (bbPtr bb) ix x
 
 getStorableAt :: Storable a => ByteBuffer -> Int -> IO a
 getStorableAt bb ix = do
-  boundCheck bb ix
+  boundCheck bb "getStorableAt" ix
   peekByteOff (bbPtr bb) ix
 
 ------------------------------------------------------------------------
 
--- indexCharOffAddr#
--- indexWideCharOffAddr#
--- indexIntOffAddr#
--- indexWordOffAddr#
--- indexAddrOffAddr#
--- indexFloatOffAddr#
--- indexDoubleOffAddr#
--- indexStablePtrOffAddr#
--- indexInt8OffAddr#
--- indexInt16OffAddr#
--- indexInt32OffAddr#
--- indexInt64OffAddr#
--- indexWord8OffAddr#
--- indexWord16OffAddr#
--- indexWord32OffAddr#
--- indexWord64OffAddr#
-
--- writeIntArray#
+-- readCharOffArray#
+-- readWideCharOffArray#
+readIntOffArrayIx :: ByteBuffer -> Int -> IO Int
+readIntOffArrayIx bb (I# ix#) = IO $ \s ->
+  case readIntArray# (bbData bb) ix# s of
```

Comment:
Should ths be called `unsafe..` or do a `boundsCheck`?

Reply:
Fixed, thanks